### ISSUE TITLE
Access Denied

### ISSUE THREAD

**Issue by michaeldoll** [ysimonx/node-red-contrib-ecoflow#1]
Hello,
when I try to establish a connection, I get
always the message Access denied.

From the support I got the following information about the API

Please check the API link below.
 
curl -X GET \
https://api.ecoflow.com/iot-service/open/api/device/queryDeviceQuota?sn=xxxxxxxxxxxxx
 \
-H 'Content-Type: application/json' \
-H 'appKey: xxxxxxxxxxxxxxxxx' \
-H 'secretKey: xxxxxxxxxxxxxxxxx'

In Ecoflow Connect I have entered the serial number of the Ecoflow Delta Max
entered. As API_key I tried the appKey and the secretKey, but I always get the message "Access denied".

**Comment by GeorgWolter** [comment]
Same problem here.

**Comment by misko85** [comment]
They have changed the API,so this extension no longer works.

**Comment by ysimonx** [comment]
Yep ... I just received this morning the new process with key & secrets

**Comment by misko85** [comment]
This sounds like a nice gift for the new year....
Many thanks for your support

**Comment by misko85** [comment]
Is there a new status? I would be happy if it should be possible to switch the outputs.

**Comment by GeorgWolter** [comment]
> Is there a new status? I would be happy if it should be possible to switch the outputs. 

Hi!
there is a new FW for the DELTA Pro which brings the access to the local API back.

**Comment by misko85** [comment]
@GeorgWolter 
Where can I find informations about the local API connection? Preffered how to do that with NodeRed.
Have a Delta 2 with actual firmware.

**Comment by ysimonx** [comment]
@misko85 @GeorgWolter 
let's do the migration ;-)

**Comment by ysimonx** [comment]
@misko85 and @GeorgWolter 
i just pushed a 0.9.6 version of the node-red ecoflow node
on node-red's palette

It is not refreshed yet on the server, i am waiting for it
so i only tested the "test.js" provided in this github repo

with a serial_number, an app_key, a secret_key

the app_key and the secret_key are provided by ecoflow's support
(for me, the email address is [support.eu@ecoflow.com](mailto:support.eu@ecoflow.com))

i have to close my computer now before the node-red palette is refreshed, may be you will be able to try it before me :)

regards

**Comment by ysimonx** [comment]
> 

Cool @misko85 

This is the only request i have been informed of : 

https://api.ecoflow.com/iot-service/open/api/device/queryDeviceQuota?sn=

if you hear about other endpoints, may be we will implement these ones ... have you got some informations ?

**Comment by GeorgWolter** [comment]
EcoFlow Cloud API: We are only getting these 4 entities from there.
But with this integration for Home Assistant we are getting are many many more entities:
https://github.com/vwt12eh8/hassio-ecoflow/issues

**Comment by ysimonx** [comment]
Ok. I don t have much time to look. But I will asap

**Comment by misko85** [comment]
About the API,I have no further information. I only know the possibility to get data via MQTT. They are very detailed, but EcoFlow sends new data every 0.5 seconds. 
The advantage of MQTT is the possibility to set parameters, like switching on or off the 220V output.
I have seen that it is also possible to set parameters via Home Assistant. But this integration uses a local API that communicates directly with the device.

**Comment by GeorgWolter** [comment]
https://youtu.be/ezn0NDc9GAY

This is what you are looking for.

**Comment by ysimonx** [comment]
> https://youtu.be/ezn0NDc9GAY
> 
> This is what you are looking for.

mqtt.ecoflow.com port 8883, interesting, thx
